templates/iloconfig.erb

```diff
@@ -1,52 +1,52 @@
-<% if autoip == true -%>
-   <% iparray = ipaddress.split('.') -%>
-   <% iloip = iparray[0] + '.' + iparray[1] + '.' + ilonet + '.' + iparray[3] -%>
-   <% gwarray = iloip.split('.') -%>
-   <% gateway = gwarray[0] + '.' + gwarray[1] + '.' + gwarray[2] + '.' + gwbit -%>
+<% if scope['autoip'] == true -%>
+   <% iparray = scope['::ipaddress'].split('.') -%>
+   <% @iloip = iparray[0] + '.' + iparray[1] + '.' + scope['ilonet'] + '.' + iparray[3] -%>
+   <% gwarray = @iloip.split('.') -%>
+   <% gateway = gwarray[0] + '.' + gwarray[1] + '.' + gwarray[2] + '.' + scope['gwbit'] -%>
 <% else -%>
-   <% iloip = ip -%>
-<% end -%> 
+   <% @iloip = scope['ip'] -%>
+<% end -%>
 
 <RIBCL VERSION="2.0">
   <LOGIN USER_LOGIN="adminname" PASSWORD="password">
-  <% if ilogen.to_i > 1 -%>
+  <% if scope['::ilogen'].to_i > 1 -%>
   <SERVER_INFO MODE="write">
-       <SERVER_NAME value ="<%= hostname %>" />
+       <SERVER_NAME value ="<%= scope['::hostname'] %>" />
   </SERVER_INFO>
-  <% end -%> 
+  <% end -%>
   <RIB_INFO MODE="write">
-<% if dhcp == false -%>
+<% if scope['dhcp'] == false -%>
     <MOD_NETWORK_SETTINGS>
       <SPEED_AUTOSELECT VALUE = "Y"/>
-<% if !shared -%>
+<% if scope['shared'] == false -%>
       <SHARED_NETWORK_PORT VALUE="N"/>
 <% end -%>
-<% if shared -%>
```

Comment:
`lookup.scope` was deprecated, vanilla scope is ok. 

https://docs.puppet.com/puppet/latest/reference/lang_template_erb.html#accessing-puppet-variables